### PR TITLE
[docs] Add perceval documentation content

### PR DESCRIPTION
**Pull request by VSevagen**
This commit adds documentation from current tutorial.
Since we're revamping old tutorial, we're keeping the
documentation in readthedocs instance.

Signed-off-by: sevagenv <sevagenv@gmail.com>

### CODE REVIEW

**Comment by vchrombie** [comment]
Thanks for opening the PR @VSevagen.

It is strange how there is no readthedocs check. I remember seeing it in the sortinghat repository. :thinking:

**Review by vchrombie — COMMENTED** [review]
I have a few comments, please let me know what do you think.

File: docs/index.rst [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
+.. toctree::
+    :maxdepth: 1
+    :caption: Perceval:
+
+    Introduction to Perceval <perceval/intro.md>
+    Retrieving git metadata <perceval/git-metadata.md>
+    Retrieving data from GitHub <perceval/git-data.md>
+    Mail archives <perceval/mail-archives.md>
+    Gerrit <perceval/gerrit.md>
+    Summary <perceval/summary.md>
```
[the remark this answers is not in the document]
Comment (VSevagen) [reply docs/index.rst:15]: Yep. Make sense. I was thinking to maybe add a link to the next docs from there, like keeping the current's tutorial's format.

File: docs/index.rst [hunk 2]
```diff
@@ -3,6 +3,17 @@
...
+
```
[the remark this answers is not in the document]
Comment (VSevagen) [reply docs/index.rst:16]: Yep. I'll add them

File: docs/index.rst [hunk 3]
```diff
@@ -3,6 +3,17 @@
...
 Indices and tables
```
[the remark this answers is not in the document]
Comment (VSevagen) [reply docs/index.rst:17]: Ok. I'll add a link to the index and modules from the sidebar.

**Comment by VSevagen** [comment]
@vchrombie I've updated the pr accordingly

**Comment by vchrombie** [comment]
Thanks for working on the changes @VSevagen. I have fixed a few issues and rebased the branch to include the latest commits.

**Review by vchrombie — APPROVED** [review]
LGTM!

**Pull request merged**